generator/src/main/java/com/graphicsfuzz/generator/mutateapi/ExpressionGenerator.java

```diff
@@ -0,0 +1,298 @@
+  private String genVarName(Value value) {
+    return Constants.GLF_PRIMITIVE_VAR + "_"
+        + value.getType().toString() + "_"
+        + value.getData().get(0).get().toString().replace(".", "_");
+  }
+
+  // TODO: Requires further implementation in the case of the negative number, i.e., -0.75 could be
+  // converted to _GLF_FLOAT_NEGATIVE_0_75.
+  private String nameFromNumber(Number number) {
+    return number.toString();
+  }
+
+  private String getFunctionName(Value value) {
+    return Constants.GLF_COMPUTE + "_"
+        + value.getType().toString().toUpperCase() + "_"
+        + nameFromNumber(value.getData().get(0).get()).replace(".", "_")
+        + "_ID_" + idGenerator.freshId();
+  }
+
+  // TODO: Handles the case of boolean value.
+  private Optional<Number> numberFromName(String string) {
+    try {
+      return Optional.of(NumberFormat.getInstance().parse(string));
+    } catch (ParseException parseException) {
+      return Optional.empty();
+    }
+  }
+
+  private Expr generateVariableFact(FactManager factManager,
+                                    FunctionDefinition functionDefinition,
+                                    Stmt stmt,
+                                    Value value,
+                                    IRandom generator) {
+    final String variableName = genVarName(value);
+    final VariableDeclInfo declInfo =
+        new VariableDeclInfo(variableName, null,
+            new Initializer(new FloatConstantExpr(value.getData().get(0).get().toString())));
+    final VariablesDeclaration variablesDecl = new VariablesDeclaration(value.getType(), declInfo);
+    functionDefinition.getBody().addStmt(new DeclarationStmt(variablesDecl));
+
+    // Store variable fact in FactManager
+    factManager.addVariableFact(declInfo, new VariableFact(variablesDecl, declInfo, value));
+    return new VariableIdentifierExpr(variableName);
+  }
+
+  private Expr generateFunction(FactManager factManager,
+                                FunctionDefinition functionDefinition,
+                                Stmt stmt,
+                                Value targetValue,
+                                IRandom generator) {
+
+    final String functionName = getFunctionName(targetValue);
+    final FactManager childFactManager = factManager.clone();
+    // Randomly generate a number of the function parameters
+    final List<ParameterDecl> params = generateParams(generator, functionDefinition,
+        childFactManager);
+    final FunctionPrototype prototype = new FunctionPrototype(functionName,
+        targetValue.getType(),
+        params);
+    final FunctionDefinition newFunctionDef = new FunctionDefinition(prototype,
+        new BlockStmt(new ArrayList<>(), false)
+    );
+    newFunctionDef.getBody().addStmt(new ReturnStmt(
+        generateExpr(childFactManager, newFunctionDef, stmt, targetValue, generator)));
+    translationUnit.addDeclaration(newFunctionDef);
```

Comment:
When you add the fact about the function to the fact manager, it should be added as a global fact.